queue_job/models/queue_job.py

```diff
@@ -398,13 +398,19 @@ def autovacuum(self):
         """
         for channel in self.env["queue.job.channel"].search([]):
             deadline = datetime.now() - timedelta(days=int(channel.removal_interval))
+            subchannels = self.env["queue.job.channel"].search(
+                [("parent_id", "=", channel.id)]
+            )
             while True:
                 jobs = self.search(
                     [
                         "|",
                         ("date_done", "<=", deadline),
                         ("date_cancelled", "<=", deadline),
+                        "|",
                         ("channel", "=", channel.complete_name),
+                        ("channel", "ilike", channel.complete_name + ".%"),
+                        ("channel", "not in", subchannels.mapped("complete_name")),
```

Comment:
It would be worth to have a test that covers several levels of hierarchy, ex:

* root (has config: 7 days)
  * root.a (has config: 3 days)
    * root.a.z (has config: 10 days)
    * root.a.y (has no config)
  * root.b (has no config)
    * root.b.z (has no config)
    * root.b.y (has no config)
    * root.b.x (has config: 1 day)
 
When cleaning the channels sequentially, the expectation should be:

1. root: clean root, root.b, root.b.z, root.b.y with 7 days
2. root.a: clean root.a and root.a.y with 3 days
3. root.a.z: clean root.a.z with 10 days
4. root.b.x: clean root.b.x with 1 day

If I'm not mistaken, by the current logic, root.a.z and root.b.x would not respect their 10 days/1 day configuration.